app/src/main/java/com/github/se/gatherspot/model/MapViewModel.kt

```diff
@@ -18,25 +19,81 @@ import com.github.se.gatherspot.model.event.Event
 import com.google.android.gms.location.LocationServices
 import com.google.android.gms.maps.model.LatLng
 import com.google.firebase.auth.FirebaseAuth
+import com.google.maps.android.compose.CameraPositionState
+import kotlin.math.PI
+import kotlin.math.cos
 
 @RequiresApi(Build.VERSION_CODES.S)
 class MapViewModel(application: Application) : AndroidViewModel(application) {
 
+  companion object {
+    fun degreeToMeters(latitudeDegrees: Double): Double {
+      // Radius of the Earth at the equator in meters
+      val earthRadiusAtEquator = 6378137.0 // in meters
+
+      // Convert latitude from degrees to radians
+      val latitudeRadians = Math.toRadians(latitudeDegrees)
+
+      // Calculate the length of a degree of latitude in meters
+      // This accounts for the Earth's curvature using the WGS-84 ellipsoid
+      // Formula: length_of_degree = (2 * PI * radius) * (cos(latitude))
+      // Where `latitude` is in radians and `radius` is the radius at that latitude
+      val radiusAtLatitude = earthRadiusAtEquator * cos(latitudeRadians)
+      val lengthOfDegree = (2 * PI * radiusAtLatitude) / 360.0
+
+      return lengthOfDegree
+    }
+
+    fun metersToDegree(latitudeDegrees: Double, meters: Double): Double {
+      // Radius of the Earth at the equator in meters
+      val earthRadiusAtEquator = 6378137.0 // in meters
```

Comment:
as earthRadiusAtEquator is used in 2 methods you may put it as a constant of the companion object